sdk/spring/spring-cloud-azure-feature-management/README.md

```diff
@@ -192,10 +192,136 @@ feature-management:
+            Start: "Wed, 01 May 2019 13:59:59 GMT",
+            End: "Mon, 01 July 2019 00:00:00 GMT"
 ```
 
+The time window can be configured to recur periodically. This can be useful for the scenarios where one may need to turn on a feature during a low or high traffic period of a day or certain days of a week. To expand the individual time window to recurring time windows, the recurrence rule should be specified in the `Recurrence` parameter.
+
+**Note:** `Start` and `End` must be both specified to enable `Recurrence`.
+
+```yaml
+feature-management:
+  feature-flags:
+    feature-v:
+      enabled-for:
+        -
+         name: TimeWindowFilter
+          parameters:
+            start: "Fri, 22 Mar 2024 20:00:00 GMT",
+            end: "Mon, 01 July 2019 00:00:00 GMT",
+            recurrence:
+              pattern:
+                type: "Daily",
+                interval: 1
+              range:
+                type: "NoEnd"
+```
+
+The `Recurrence` settings is made up of two parts: `Pattern` (how often the time window will repeat) and `Range` (for how long the recurrence pattern will repeat).
+#### Recurrence Pattern
+
+There are two possible recurrence pattern types: `Daily` and `Weekly`. For example, a time window could repeat "every day", "every 3 days", "every Monday" or "on Friday per 2 weeks".
+
+Depending on the type, certain fields of the `Pattern` are required, optional, or ignored.
+
+- `Daily`
+
+  The daily recurrence pattern causes the time window to repeat based on a number of days between each occurrence.
+
+  | Property | Relevance | Description |
+      |----------|-----------|-------------|
+  | **Type** | Required | Must be set to `Daily`. |
+  | **Interval** | Optional | Specifies the number of days between each occurrence. Default value is 1. |
+
+- `Weekly`
+
+  The weekly recurrence pattern causes the time window to repeat on the same day or days of the week, based on the number of weeks between each set of occurrences.
+
+  | Property | Relevance | Description |
+      |----------|-----------|-------------|
+  | **Type** | Required | Must be set to `Weekly`. |
+  | **DaysOfWeek** | Required | Specifies on which day(s) of the week the event occurs. |
+  | **Interval** | Optional | Specifies the number of weeks between each set of occurrences. Default value is 1. |
+  | **FirstDayOfWeek** | Optional | Specifies which day is considered the first day of the week. Default value is `Sunday`. |
+
+  The following example will repeat the time window every other Monday and Tuesday
+
+    ```yaml
+    pattern:
+        type: "Weekly",
+        interval: 1,
+        daysOfWeek: 
+          - Monday
+          - TuesDay
+    ```
+  **Note:** `Start` must be a valid first occurrence which fits the recurrence pattern. Additionally, the duration of the time window cannot be longer than how frequently it occurs. For example, it is invalid to have a 25-hour time window recur every day.
```

Comment:
> If I want to repeat on Monday/Wednesday/Friday from 2pm-10pm for example what values must my Start and End be?

Your `Start` should be 2pm on Monday/Wednesday/Friday. Your `End` should be 10pm on the same date as `Start`.

The reason why we require `Start` to be a valid first occurrence is after we decide to follow the Outlook schema, we found the startDate and recurrenceTimeZone in Recurrence.Range can be simplified since the information has been included in `Start`.

For more context, you can refer to this discussion.
https://github.com/microsoft/FeatureManagement-Dotnet/pull/256#issuecomment-1732129214

Reply:
> What if my time period starts on one day but ends on another because timezones? 

It doesn't matter. We don't care about the timezone of `End`, we only use `End` to calculate time window duration (`End` - `Start`). The timezone of `Start` decides which timezone the recurrence pattern will be applied to.

Reply:
A time window could run for 25 hours due to daylight savings time.

Reply:
> A time window could run for 25 hours due to daylight savings time.

This could not happen. For original timewindowfilter, this case will not happen.
The `Start` and `End` is two explicit datetime/timestamps and the duration of the timewindow is fixed.

Reply:
What about the first day is the daylight savings time day.

Reply:
It follows the time zone of the `Start`.